packages/amplify-prompts/src/stopwatch.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Class to record duration of interactions
+ */
+export class Stopwatch {
+    private _currentState : StopwatchState = StopwatchState.IDLE
+    private _slices: Slice [] = []
+
+    start = (): void => {
+      this._currentState = StopwatchState.RUNNING;
+      const now = Date.now();
+      this._slices.push({
+        start: now,
+      });
+    }
+
+    /**
+     * stops the time and reset
+     */
+    stop = () : void => {
+      this._currentState = StopwatchState.IDLE;
+      this._slices.length = 0;
+    }
+
+    /**
+     * pauses the time call start to resume
+     */
+    pause = (): void => {
+      if (this._currentState === StopwatchState.IDLE) {
+        throw new Error('Cannot pause no timer running');
```

Comment:
Will this terminate/error out the entire CLI process if the state gets out of sync? I can't really tell how it catches at the prompter level, but it seems like the stop watch should gracefully handle and persist the error on our end without stopping the developer's CLI command.

Reply:
+1

Reply:
Sounds good let me make that change